[web-components] refactor color recipes for DI (#18199)
microsoft/fluentui: packages/web-components/src/color-vNext/README.md

> # FAST Color Recipes

Reviewer: ```suggestion
# FluentUI Color Recipes
```

> Color recipes are named colors who's value is algorithmically defined from a variety of inputs. `@microsoft/fast-components` relies on these recipes heavily to achieve expressive theming options while maintaining color accessability targets.

Reviewer: ```suggestion
Color recipes are named colors who's value is algorithmically defined from a variety of inputs. `@fluentui/web-components` relies on these recipes heavily to achieve expressive theming options while maintaining color accessibility targets.
```

## Swatch

A Swatch is a representation of a color that has a `relativeLuminance` value and a method to convert the swatch to a color string. It is used by recipes to determine which colors to use for UI.

### SwatchRGB

A concrete implementation of `Swatch`, it is a swatch with red, green, and blue 64bit color channels .

**Example: Creating a SwatchRGB**

```ts

> import { SwatchRGB } from '@microsoft/fast-components';

Reviewer: ```suggestion
import { SwatchRGB } from '@fluentui/web-components';
```
